### PR TITLE
Natron: init at 1.2.1

### PR DESCRIPTION
**Pull request by puffnfresh**
Wasn't an easy thing to package but it runs great. Was able to generate some images, animate them with some keyframes and render to an MP4 file.

### CODE REVIEW

**Comment by bjornfor** [comment]
Please use lower-case directory names (thinking of "pkgs/applications/video/Natron/").

**Comment by bjornfor** [comment]
And lower-case package name: Natron -> natron.

**Comment by davidak** [comment]
i tried to build on NixOS x86_64 with `nox-review pr 9846`:

```
[...]
From https://github.com/devernay/openfx-supportext
 * [new branch]      master     -> origin/master
 * [new tag]         Natron-0.9.6 -> Natron-0.9.6
Switched to a new branch 'fetchgit'
git revision is 0c079bafeab5750e6e26086c67c5947a0ce70097
git human-readable version is 1.9.0
Commit date is 2015-08-05 18:41:49 +0200
Deleted remote-tracking branch origin/master (was c68738a).
Deleted remote-tracking branch origin/gh-pages (was c71528b).
Deleted remote-tracking branch origin/master (was 6eb0a50).
output path ‘/nix/store/nllb1gijq8qv8857dgx6h2m17bw5gadm-openfx-misc’ should have r:sha256 hash ‘0zbwyxblkcyyyj45szhr2vhrbjwlpgcxm24pvp5048ns5rwxwbbh’, instead has ‘0iazhaycbcsv2vxzg9slh32bsvz1yj55453szybac97cm3g70i3k’
```

**Comment by vcunat** [comment]
That's what `fetchgit` from github does &ndash; the stuff served changes slightly over time. `fetchzip` and `fetchFromGitHub` were introduced exactly to fight that.

File: pkgs/applications/video/natron/default.nix [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  openexr, ffmpeg, opencolorio, openimageio, imagemagick }:
+
+let
+  version = "1.2.1";
+  CImg = fetchurl {
+    url = https://raw.githubusercontent.com/dtschump/CImg/d89fbcf399b3663ce3ebac2ee387642fd64014b6/CImg.h;
+    sha256 = "065sbx8dkywkrimc0l0vpq7dy7m9y6jlaq84x5ngys661j2nnslv";
+  };
+  plugins = {
+    misc = fetchgit {
```
Comment (jagajaga) [inline pkgs/applications/video/natron/default.nix]: Please, can you change every `fetchgit` to `fetchFromGithub` if it is possible?

Comment (puffnfresh) [inline pkgs/applications/video/natron/default.nix]: Natron has quite a few git submodules and so I wasn't about to use fetchFromGitHub. Do I need to include each submodule as a separate fetch?

Comment (jagajaga) [inline pkgs/applications/video/natron/default.nix]: If there are submodules in a repository then use `fetchgit` otherwise use `fetchFromGithub` :)

**Comment by joachifm** [comment]
@puffnfresh ping?

**Comment by garbas** [comment]
@puffnfresh can you rebase on master?

**Comment by garbas** [comment]
i'm closing this PR since package in this PR is not the latest and there has been almost one year of inactivity.

**Pull request closed without merging**